src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -508,24 +508,18 @@ private bool ReplacementBreaksCollectionInitializerAddMethod(ExpressionSyntax or
             return !SymbolsAreCompatible(originalSymbol, newSymbol);
         }
 
-        protected override bool IsInvocableExpression(SyntaxNode node)
+        protected override bool ExpressionMightReferenceMember(SyntaxNode node)
         {
-            if (node.IsKind(SyntaxKind.InvocationExpression) ||
+            return node.IsKind(SyntaxKind.InvocationExpression) ||
                 node.IsKind(SyntaxKind.ObjectCreationExpression) ||
-                node.IsKind(SyntaxKind.ElementAccessExpression))
-            {
-                return true;
-            }
-
-            if (node.IsKind(SyntaxKind.SimpleMemberAccessExpression) &&
-                !node.IsParentKind(SyntaxKind.InvocationExpression) &&
-                !node.IsParentKind(SyntaxKind.ObjectCreationExpression) &&
-                !node.IsParentKind(SyntaxKind.ElementAccessExpression))
-            {
-                return true;
-            }
+                node.IsKind(SyntaxKind.ElementAccessExpression) ||
+                node.IsKind(SyntaxKind.SimpleMemberAccessExpression);
+        }
 
-            return false;
+        protected override bool HasArgumentList(ExpressionSyntax expression)
+        {
+            var argumentsList = GetArgumentList(expression);
+            return argumentsList != null;
```

Comment:
this code is duplicated with VB.  it's unclear to me why it's better than what we had before.